### PR TITLE
Add word similarity dashboard

### PR DESCRIPTION
**Pull request by jcorrado76**
This PR cleans up some of the utility code for generating the word similarity, and adds a dashboard that will let users add new words and visualize their embeddings on the fly.

The dashboard looks like this:
image

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    entry: hadolint
+    description: Run hadolint to lint Dockerfiles
+    language: system
+    types: ["dockerfile"]
+- repo: https://github.com/zahorniak/pre-commit-circleci.git
+  rev: v0.6
+  hooks:
+    - id: circleci_validate
+      description: Validate `.circleci/config.yml` file
+# - repo: https://github.com/sourcery-ai/sourcery
```
Comment (abhishek-kumar) [inline .pre-commit-config.yaml:39]: Can we delete the commented lines?

File: 2_large_language_models/README.md [hunk 2]
```diff
@@ -1 +1,30 @@
 # Weeks 4 & 5 - Large Language Models
+
+The first ML project we'll be developing in the LLM section is a word similarity machine.
```
Comment (abhishek-kumar) [inline 2_large_language_models/README.md:3]: Word similarity machine - nice!

File: 2_large_language_models/llm_masterclass/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+hello_world = "hello world"
```
Comment (abhishek-kumar) [inline 2_large_language_models/llm_masterclass/__init__.py:1]: Is this needed?

File: 2_large_language_models/ml_project_1_similar_words/app.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+words_to_start = [
+    "Paris",
+    "Python",
+    "Sunday",
+    "Tolstoy",
+    "Twitter",
+    "bachelor",
+    "delivery",
+    "election",
```
Comment (abhishek-kumar) [inline 2_large_language_models/ml_project_1_similar_words/app.py:23]: Interesting set of words!

File: 2_large_language_models/README.md [hunk 5]
```diff
@@ -1 +1,30 @@
...
+
+The second project will be to create a system for predicting sentiment analysis, given a sentence.
+
+## Project 1 - Word Similarity
+
+For this week, we'll go through the process of creating a word similarity machine. This is a system that can take two words and determine how similar they are. For example, the words "cat" and "dog" are very similar, while the words "cat" and "banana" are not.
```
Comment (abhishek-kumar) [inline 2_large_language_models/README.md:9]: Do you want to use the term "application" or "service" instead of "machine"?

File: README.md [hunk 6]
```diff
@@ -1,17 +1,16 @@
...
+The structure of the repo consists of a directory for each section. Each section contains multiple directories for the hands-on projects:
 
-* 0_introduction_to_tensorflow (Week 1)
-* 1_model_deployment (Week 1)
-* 2_large_language_models (Weeks 4 & 5)
-* 3_computer_vision (Weeks 6 & 7)
+* `0_introduction_to_tensorflow` (Week 1)
+* `1_model_deployment` (Week 1)
+* `2_large_language_models` (Weeks 4 & 5)
+* `3_computer_vision` (Weeks 6 & 7)
```
Comment (abhishek-kumar) [inline README.md:10]: You can actually remove the week information from here. Let them stay as 4 separate modules numbered from 0 to 3..

**Review by abhishek-kumar — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jcorrado76) [reply .pre-commit-config.yaml:39]: Done

File: 2_large_language_models/llm_masterclass/__init__.py (hunk 3 above)
Comment (jcorrado76) [reply 2_large_language_models/llm_masterclass/__init__.py:1]: I have it in here as an example of how to create a Python package, and then install it in editable mode. In the tests, I show how you'd import this hello world, and assert it has the correct value.

File: 2_large_language_models/README.md (hunk 5 above)
Comment (jcorrado76) [reply 2_large_language_models/README.md:9]: Switched to application

**Pull request merged**